### ISSUE TITLE
First class support for distinct?

### ISSUE THREAD

**Issue by ajselvig** [nviennot/nobrainer#174]
This is easy enough to do in ReQL, just wondering if it's worth offering NoBrainer syntax for it:

```ruby
NoBrainer.run { |r| r.table("model").distinct({"index" => "foo"})
```

Maybe something like: 

```ruby
Model.distinct(index: :foo)
```

**Comment by nviennot** [comment]
hum.. dinstinct kinda break down the notion of objects mapped to documents. I'm not sure I can do an API that makes sense. distinct() kinda makes sense only when used with .raw() right?

**Comment by ajselvig** [comment]
Yeah, there wouldn't be any mapping, just returning a raw value (array). So more like Model.count(). 

Like I said, it's easy enough right now, so it's not a huge deal.

**Comment by zedtux** [comment]
Shall we close this issue?
